variables.tf

```
@@ -5,7 +5,27 @@ variable "iam_source_json_url" {
 }
 
 variable "iam_policy_statements" {
-  type        = any
+  type = object({
+    sid           = optional(string, "")
+    effect        = optional(string, "")
+    actions       = optional(list(string), [])
+    not_actions   = optional(list(string), [])
+    resources     = optional(list(string), [])
+    not_resources = optional(list(string), [])
```

- Add `version` here and in `main.tf`.
- Move statement components to list of objects.
- Inputs that are not going to be expanded into blocks by `dynamic` should default to `null`.
- In `main.tf`, replace lookups like

```hcl
lookup(statement.value, "sid", statement.key)
```

With direct references

```hcl
statement.value.sid
```

```suggestion
  type = object({
    version       = optional(string, null)
    statement = list(object({
      sid           = optional(string, null)
      effect        = optional(string, null)
      actions       = optional(list(string), null)
      not_actions   = optional(list(string), null)
      resources     = optional(list(string), null)
      not_resources = optional(list(string), null)
    })), [])
```

---

@Nuru thanks a lot for your review!
* Version arguments should be set for data source `aws_iam_policy_document`, so I added it as a separate variable. 
* I updated statements to the map of objects - that's what is expected [in the example](https://github.com/cloudposse/terraform-aws-iam-policy/blob/main/examples/complete/fixtures.us-east-2.tfvars#L58).
* Inputs were updated.
* Lookups were replaced👍 

Please let me know your thoughts.

---

@gberenice I forgot we want this to be backward compatible and that means the input is a map. I will take it from here. Thank you very much for your contributions.

---

@Nuru sounds good, thank you!